### PR TITLE
updated rock impersonation parameter api call

### PR DESCRIPTION
**Pull request by delianides**
- updated `impersonationParameter` schema to accept `expireDateTime`, `pageId`, `usageLimit` as arguments.
- updated the resolver to pass in addition args based on the api docs.
- passed the above args with `PersonId` as querystrings to the rock API.

# Test Queries

```graphql
  query PersonData {
    person: currentPerson {
      id
      firstName
      nickName
      impersonationParameter
    }
  }
```
and

```graphql
  query PersonData {
    person: currentPerson {
      id
      firstName
      nickName
      impersonationParameter(expireDateTime: "2017-09-03T17:56:26-04:00", pageId: 100, usageLimit: 2)
    }
  }
```

Note: you can leave off any of the querystring params (or empty entirely) and a IP will still be created.

### CODE REVIEW

File: src/rock/models/people/model.js [hunk 1]
```diff
@@ -153,13 +155,29 @@ export class Person extends Rock {
...
-          where: { PersonId: `${id}` }
-        })
+          where: { PersonId: `${id}` },
+        }),
     );
   }
 
-  async getIP(id) {
-    return PersonTable.fetch("GET", `GetSearchDetails/${id}`);
+  async getIP(id, args) {
```
Comment (delianides) [inline src/rock/models/people/model.js:163]: main model method here

File: src/rock/models/people/resolver.js [hunk 2]
```diff
@@ -77,14 +78,14 @@ export default {
     firstName: ({ FirstName }) => FirstName,
     lastName: ({ LastName }) => LastName,
     nickName: ({ NickName }) => NickName,
-    impersonationParameter: ({ Id }, _, { models, person }) => {
+    impersonationParameter: ({ Id }, args, { models, person }) => {
```
Comment (delianides) [inline src/rock/models/people/resolver.js:81]: resolver method

File: src/rock/models/people/schema.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ export default [
     birthMonth: Int
     birthYear: Int
     email: String
-    impersonationParameter: String
+    impersonationParameter(expireDateTime: String, pageId: Int, usageLimit: Int): String
```
Comment (delianides) [inline src/rock/models/people/schema.js:28]: schema

**Comment by newspringweb** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Review by richarddubay — APPROVED** [review]

**Pull request merged**